src/platform/plugins/shared/discover/public/application/main/components/top_nav/discover_topnav.tsx

```diff
@@ -211,6 +216,45 @@ export const DiscoverTopNav = ({
     [searchBarCustomization?.CustomSearchBar, navigation.ui.AggregateQueryTopNavMenu]
   );
 
+  const searchDraftUiState = useCurrentTabSelector((state) => state.uiState.searchDraft);
+  const setSearchDraftUiState = useCurrentTabAction(internalStateActions.setSearchDraftUiState);
+  const onDraftChange = useCallback(
+    (draft: UnifiedSearchDraft | null) => {
+      dispatch(
+        setSearchDraftUiState({
+          searchDraftUiState: {
+            query: draft?.query ?? undefined,
+            dateRangeFrom: draft?.dateRangeFrom ?? undefined,
+            dateRangeTo: draft?.dateRangeTo ?? undefined,
+          },
+        })
+      );
+    },
+    [dispatch, setSearchDraftUiState]
+  );
+
+  const draft = useMemo(() => {
+    const draftState: Partial<UnifiedSearchDraft> = {};
+
+    if (searchDraftUiState?.query && !isEqual(searchDraftUiState.query, query)) {
+      if (isOfAggregateQueryType(searchDraftUiState.query) !== isOfAggregateQueryType(query)) {
```

Comment:
I wonder if there's a way to incorporate this directly into Unified Search so consumers don't need to account for it themselves?

Reply:
Moved to https://github.com/elastic/kibana/pull/227298/commits/d582bf12b9090a3ab189456e00c085ce5bd67690